Tests/SonarRunner.Shim.Tests/PropertiesWriterTest.cs

```diff
@@ -125,6 +127,39 @@ public void PropertiesWriterToString()
         }
 
         [TestMethod]
+        public void PropertiesWriter_ComputeProjectBaseDir()
+        {
+            AssertHasProjectBaseDir(new List<string> { @"c:\src\A\1.proj" }, @"c:\src\A", @"z:\");
+            AssertHasProjectBaseDir(new List<string> { @"c:\src\A\1.proj", @"c:\src\A\2.proj" }, @"c:\src\A", @"z:\");
+            AssertHasProjectBaseDir(new List<string> { @"c:\src\A\1.proj", @"c:\src\B\2.proj" }, @"c:\src", @"z:\");
+            AssertHasProjectBaseDir(new List<string> { @"c:\common\src\A\1.proj", @"c:\common\src\B\2.proj" }, @"c:\common\src", @"z:\");
+            AssertHasProjectBaseDir(new List<string> { @"c:\common\src\A\1.proj", @"c:\common\src\B\2.proj", @"c:\outside\src\C\3.proj" }, @"c:", @"z:\");
+            AssertHasProjectBaseDir(new List<string> { @"c:\src\A\1.proj", @"d:\src\B\2.proj" }, @"z:", @"z:");
+        }
+
+        private void AssertHasProjectBaseDir(IEnumerable<string> projectPaths, string expectedProjectDir, string fallback)
```

Comment:
You could use "params" to pass in a variable number of arguments e.g.
public static void AssertHasProjectBaseDir(string expected, string fallback, params string[] projectPaths).